src/main/java/org/opentripplanner/framework/collection/MapUtils.java

```diff
@@ -28,4 +33,21 @@ public static <U> void addToMapSet(TLongObjectMap<Set<U>> mapSet, long key, U va
   public static <S, T> List<T> mapToList(Collection<S> entities, Function<S, T> mapper) {
     return entities == null ? null : entities.stream().map(mapper).collect(Collectors.toList());
   }
+
+  /**
+   * Takes a list of maps and returns the union of all of them.
+   * <p>
+   * If there are duplicate keys then the one from the earlier argument overwrites the value in a
+   * later argument.
+   */
+  @SafeVarargs
+  public static <K, V> Map<K, V> combine(Map<K, V>... maps) {
+    var ret = new HashMap<K, V>();
+    // need to put it into a mutable list, so that Collections.reverse works
+    var entries = new ArrayList<>(Arrays.stream(maps).flatMap(m -> m.entrySet().stream()).toList());
+    // we reverse the entries so that if there are duplicate keys, the earlier method arguments take precedence
+    Collections.reverse(entries);
```

Comment:
This seems from the usage to be wrong, I would let the last element overwrite the first, so if:

(DEFAULTS, SPECIFIC) would end up using the SPECIFIC value if it exist.

Reply:
I have reversed the order.